src/blockdata/script.rs

```diff
@@ -216,6 +217,16 @@ impl Script {
     /// Creates a new empty script
     pub fn new() -> Script { Script(vec![].into_boxed_slice()) }
 
+    /// Returns 160-bit hash of the script
+    pub fn script_hash(&self) -> ScriptHash {
+        ScriptHash::hash(&self.serialize())
```

Comment:
This makes an unnecessary allocation. Please serialize directly into the hashengine.

Reply:
This one is actually an allocation, and needs to be addressed

Reply:
Never mind, it is fixed in a later commit.